pyproject.toml

```diff
@@ -0,0 +1,67 @@
+[tool.poetry]
+name = "lib-ml-REMLA10-2024"
+version = "0.0.0"
+description = "Pre-processing logic for some ML work."
+authors = ["Jan <j2000.vdm@gmail.com>", "Shayan Ramezani <s.ramezani@student.tudelft.nl>", "Remi Lejeune <R.J.Lejeune@student.tudelft.nl>", "Michael Chan <J.M.Chan@student.tudelft.nl>"]
+license = "MIT"
+readme = "README.md"
+packages = [
+    { include = "lib_ml_remla", from = "src" }
+]
+
+[tool.poetry.dependencies]
+python = "~3.11"
+tensorflow-io-gcs-filesystem = [
+    {platform = "darwin", version = "~0.36.0"},
+    {platform = "linux", version = "~0.31.0"},
+    {platform = "win32", version = "~0.31.0"},
+]
+tensorflow = {version = "~2.16.1" }
+tensorflow-intel = { version = "~2.16.1", platform = "win32" }
+keras = "3.3.3"
+scikit-learn = "1.2.2"
+
+
+[tool.poetry.group.dev]
+optional = true
+
+[tool.poetry.group.dev.dependencies]
+bump-my-version = "^0.21.0"
+pytest = "^8.0.0"
+numpy = "^1.26"
```

Comment:
Is there a reason why numpy is an optional dev dependency rather than a mandatory one?

Reply:
because it's only used in testing and not the core logic

Reply:
But its imported by preprocessing.py right